src/wp-admin/includes/class-wp-comments-list-table.php

```diff
@@ -942,9 +942,25 @@ public function column_author( $comment ) {
 			$author_url_display = wp_html_excerpt( $author_url_display, 49, '&hellip;' );
 		}
 
-		echo '<strong>';
-		comment_author( $comment );
-		echo '</strong><br />';
+		$author_name = get_comment_author( $comment );
+
+		if ( $comment->user_id ) {
+			$author_profile = admin_url( 'user-edit.php?user_id=' . $comment->user_id );
+
+			// Link the author's name to their profile page.
+			printf(
+				'<a href="%s"><strong>%s</strong></a><br />',
+				esc_attr( $author_profile ),
+				$author_name
+			);
+		} else {
+			// Indicate an unregistered author.
+			printf(
+				'<strong>%s</strong> (%s)<br />',
+				$author_name,
+				__( 'unregistered' )
+			);
```

Comment:
Should this be hookable?

```php
$default = '(' . __( 'unregistered' ) . ')';

/**
 * Filters the text shown on the Comments screen to indicate
 * that a comment author is not registered on the site.
 *
 * @since x.x.x
 *
 * @param string     $unregistered_text The text to indicate that a comment author is not
 *                                      registered on the site. Default '(unregistered)'.
 * @param WP_Comment $comment           The comment.
 */
$unregistered_text = apply_filters( 'comment_author_unregistered', $default, $comment );

printf( '<strong>%s</strong> %s<br />', $author_name, $unregistered_text );
```